typify-impl/src/lib.rs

```diff
@@ -221,6 +227,39 @@ impl TypeSpace {
 
         // TODO Look for containment cycles that we need to break with a Box<T>
 
+        // While we aren't yet handling the general case of type containment
+        // cycles, it's not that bad to look at trivial cycles (A->A).
```

Comment:
```suggestion
        // cycles, it's not that bad to look at trivial cycles (A->A) (including A->Option<A>).
```

Reply:
Actually, I think this comment just needs a rewhack based on what's going on below now...

Reply:
rewhacked as part of 3459a0c827ae97b68eb19352d4b39c8edaa56488